hs_geographic_feature_resource/models.py

```diff
@@ -0,0 +1,390 @@
+                if key in ('geometryType', 'featureCount'):
+                    setattr(geom_info, key, value)
+            geom_info.save()
+        else:
+            raise ObjectDoesNotExist("No GeometryInformation element can be found for the provided id: %s" % kwargs['id'])
+
+    @classmethod
+    def remove(cls, element_id):
+        geom_info = GeometryInformation.objects.get(id=element_id)
+        if geom_info:
+            geom_info.delete()
+        else:
+            raise ObjectDoesNotExist("GeometryInformation element of the geographic feature resource cannot be deleted: %d."%element_id)
+
+
+# Define the Geographic Feature
+class GeographicFeatureResource(BaseResource):
+    objects = ResourceManager("GeographicFeatureResource")
+    @property
+    def metadata(self):
+        md = GeographicFeatureMetaData()
+        return self._get_metadata(md)
+
+    @classmethod
+    def get_supported_upload_file_types(cls):
+    # See Shapefile format: http://resources.arcgis.com/en/help/main/10.2/index.html#//005600000003000000
+        return (".zip", ".shp", ".shx", ".dbf", ".prj", ".sbx", ".sbn", ".cpg", ".xml", ".fbn", ".fbx", ".ain", ".aih", ".atx", ".ixs", ".mxs")
+
+    @classmethod
+    def can_have_multiple_files(cls):
+        # can have more than one files
+        return True
+
+    class Meta:
+        verbose_name = 'Geographic Feature (ESRI Shapefiles)'
+        proxy = True
+
+processor_for(GeographicFeatureResource)(resource_processor)
+
+# define the GeographicFeatureMetaData metadata
+class GeographicFeatureMetaData(CoreMetaData):
+    geometryinformation = generic.GenericRelation(GeometryInformation)
+    fieldinformation = generic.GenericRelation(FieldInformation)
+    originalcoverage = generic.GenericRelation(OriginalCoverage)
+    originalfileinfo = generic.GenericRelation(OriginalFileInfo)
+
+    @classmethod
+    def get_supported_element_names(cls):
+        # get the names of all core metadata elements
+        elements = super(GeographicFeatureMetaData, cls).get_supported_element_names()
+        # add the name of any additional element to the list
+        elements.append('FieldInformation')
+        elements.append('OriginalCoverage')
+        elements.append('GeometryInformation')
+        elements.append('OriginalFileInfo')
+        return elements
+
+    def has_all_required_elements(self):
+        if not super(GeographicFeatureMetaData, self).has_all_required_elements():  # check required meta
+            return False
+        if not self.originalfileinfo.all().first():
+            return False
+        if not self.fieldinformation.all().first():
+            return False
+        if not self.originalcoverage.all().first():
+            return False
+        if not self.geometryinformation.all().first():
+            return False
+        if not (self.coverages.all().filter(type='box').first() or self.coverages.all().filter(type='point').first()):
+            return False
+        return True
```

Comment:
@ironfroggy  Looks like every apps should implement these two functions. I am guessing the intention of this design is to avoid redundant checks. The first func (has_all_required_elements) is an initial try, once a missing required metadata is detected,  then the second func (get_required_missing_elements) will be called to get the names of all missing metadata. Maybe 
@pkdash @hyi  can help me explain this?